google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -180,7 +203,8 @@ def lease(self, items: Sequence[requests.LeaseRequest]) -> None:
         Args:
             items: The items to lease.
         """
-        self._manager.leaser.add(items)
+        leaser = cast("Leaser", self._manager.leaser)
```

Comment:
Seems like a signal that `StreamingPullManager.leaser` needs better typing.  If it is `Optional`, then an `assert self._manager.leaser is not None` would work here.

Reply:
Leaser can be `None`, thus the `cast()` to avoid the warning. But I like the `assert` trick better, it's much more clear what's going on here.